### ISSUE TITLE
Fails during import due to numpy deprecating np.float

### ISSUE THREAD

**Issue by iboates** [pysal/spopt#341]
I just installed spopt for the first time and found that it would not import, throwing:

```
AttributeError: module 'numpy' has no attribute 'float'
```

I have been seeing this error a lot recently when working with older code it is caused by numpy recently going from deprecation to complete removal of `np.float` [(this stackoverflow answer explains quite clearly)](https://stackoverflow.com/questions/74844262/how-to-solve-error-numpy-has-no-attribute-float-in-python)

I have been able to fix it in the meantime with a brute force global replacement of `np.float` with `float`, which, actually should be sufficient since the intent of numpy was to just use the built-in `float` rather than relying on their own explicit definition.

I don't really know if such a coarse replacement would be sufficient for the library, but I can make a PR from it, if desired

A patch for the current release would probably be to peg the numpy requirements to versions that are pre-removal:

```
numpy>=1.21,<1.24
```

**Comment by jGaboardi** [comment]
@iboates Thanks for the report. This is actually fixed in `main` and will be part of the `v0.5.0` release. You can install `v0.5.0rc1` [via pip](https://pypi.org/project/spopt/0.5.0rc1/#description) and it should also be [available via soon](https://github.com/conda-forge/spopt-feedstock/pull/12).

**Comment by jGaboardi** [comment]
New version out resolves this issue.
